f8a_worker/workers/graph_populator.py

```diff
@@ -0,0 +1,242 @@
+import logging
+import re
+import time
+from datetime import datetime
+
+logger = logging.getLogger(__name__)
+
+# Compile patterns to clean a string (used for safety against GraphQL injection attack)
+danger_word_list = ["drop\(\)", "V\(\)", "count\(\)"]
+remove = '|'.join(danger_word_list)
+pattern = re.compile(r'(' + remove + ')', re.IGNORECASE)
+pattern_to_save = '[^\w\*\.Xx\-\>\=\<\~\^\|\/\:]'
+pattern_n2_remove = re.compile(pattern_to_save)
+
+
+class GraphPopulator(object):
+    @classmethod
+    def str_value_cleaner(cls, safe_string):
+        """
+        Clean up the safe_string to remove suspicious words,
+        and special characters.
+        """
+        result = ''
+        if isinstance(safe_string, str):  # can raise an exception as well
+            temp_str = pattern.sub('', safe_string)
+            result = pattern_n2_remove.sub('', temp_str)
+        return result
+
+    @classmethod
+    def construct_version_query(cls, input_json):
+        description = ''
+        try:
+            if len(input_json.get('analyses', {}).get('metadata', {}).get('details')) > 0:
+                description = input_json.get('analyses').get('metadata').get('details')[0].get('description', '')
+                description = cls.str_value_cleaner(description).replace("'", "\\'")
+        except:
+            # we pass and move forward without description
```

Comment:
maybe debug log here would be nice that we skipped description.

Reply:
I feel not the need for.